### PR TITLE
Add support for perf counter collection in DEX

### PR DESCRIPTION
**Pull request by jbobba**
Supports --timing_detail in nbench.

### CODE REVIEW

File: src/ngraph/runtime/cpu/cpu_external_function.cpp [hunk 1]
```diff
@@ -414,6 +412,12 @@ void runtime::cpu::CPU_ExternalFunction::compile()
     writer << "// Generated by the nGraph CPU backend\n";
     if (m_use_tbb)
     {
+        if (runtime::cpu::IsTracingEnabled() || m_emit_timing)
+        {
+            NGRAPH_WARN
```
Comment (rkimballn1) [inline src/ngraph/runtime/cpu/cpu_external_function.cpp]: Libraries should not print things. Remove the warning or find some way to get this information to an app like nbench.

File: src/ngraph/runtime/cpu/cpu_external_function.cpp [hunk 2]
```diff
@@ -1143,6 +1147,13 @@ void runtime::cpu::CPU_ExternalFunction::build()
     {
         return;
     }
+
+    if (m_use_tbb && (runtime::cpu::IsTracingEnabled() || m_emit_timing))
+    {
+        NGRAPH_WARN << "CPU Backend: Tracing and performance breakdowns might not be accurate with "
```
Comment (rkimballn1) [inline src/ngraph/runtime/cpu/cpu_external_function.cpp]: Remove warning

File: src/ngraph/runtime/cpu/cpu_external_function.cpp [hunk 3]
```diff
@@ -1615,6 +1663,49 @@ const runtime::cpu::LayoutDescriptorPtrs&
     return result_layout_descriptors;
 }
 
+const vector<runtime::PerformanceCounter>& runtime::cpu::CPU_ExternalFunction::get_perf_counters()
+{
+    if (m_direct_execution)
+    {
+        return m_perf_counters;
+    }
+#if !defined(NGRAPH_DEX_ONLY)
```
Comment (rkimballn1) [inline src/ngraph/runtime/cpu/cpu_external_function.cpp]: I don't understand why perf counters can't be used with dex.

Comment (jbobba) [reply src/ngraph/runtime/cpu/cpu_external_function.cpp]: Not sure if I follow your concern. They are being added to DEX with this PR.

File: src/ngraph/runtime/cpu/cpu_external_function.cpp (hunk 1 above)
Comment (jbobba) [reply src/ngraph/runtime/cpu/cpu_external_function.cpp]: This is only going to kick in for ngraph developers. Dont want to spend the time implementing the plumbing for this. Will make this an assert for now.

**Pull request merged**